src/analysis/object.rs

```diff
@@ -49,6 +49,32 @@ impl Info {
             || self.notify_signals.iter().any(|s| s.trampoline.is_ok())
     }
 
+    /// Whether we should generate an impl block for this object
+    /// We don't generate an impl block if the type doesn't have any of the followings:
+    /// - Constructors / Functions / Builder properties (no build function)
+    /// - Is a final type & doesn't have either methods / properties / child properties / signals
+    pub fn should_generate_impl_block(&self) -> bool {
+        self.has_constructors
+            || has_builder_properties(&self.builder_properties)
+            || !self.need_generate_trait()
+                && !(self.methods().is_empty()
+                    && self.properties.is_empty()
+                    && self.child_properties.is_empty()
+                    && self.signals.is_empty())
+            || self.has_functions
+    }
+
+    pub fn need_generate_inherent(&self) -> bool {
+        self.has_constructors
+            || self.has_functions
+            || !self.need_generate_trait()
+            || has_builder_properties(&self.builder_properties)
+    }
+
+    pub fn need_generate_trait(&self) -> bool {
```

Comment:
Tbh I don't think this should exist as the `generate_trait` is public; or do we want stuff to be private & create getters for them when needed?

Reply:
Sometimes using fields, sometimes getters seems a bit confusing. That looks like it needs a bit more cleanup than would be in the scope of this PR.

Reply:
Indeed, just wondering in general in case I get bored someday :-)